diff --git a/src/Network/AWS/Internal/Logger.hs b/src/Network/AWS/Internal/Logger.hs
index aaaaddb..c741abd 100644
--- a/src/Network/AWS/Internal/Logger.hs
+++ b/src/Network/AWS/Internal/Logger.hs
@@ -29,7 +29,7 @@ module Network.AWS.Internal.Logger
 
 import           Control.Monad
 import           Control.Monad.IO.Class
-import qualified Data.ByteString.Lazy.Builder as Build
+import qualified Data.ByteString.Builder as Build
 import           Data.Monoid
 import           Network.AWS.Data.Log
 import           Network.AWS.Types
